crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -142,67 +348,106 @@ impl<I: Iterator<Item = Result<(OutputHeader, Output)>>> Iterator for MergingIte
                 Ok((output_header_with_balance.output_header, basic.into()))
             })
     }
+
+    fn print_metrics(&self) {
+        if self.with_metrics {
+            info!(
+                "Number of UnlockedVestingOutputFilter modified outputs: {}",
+                self.modified_outputs.len()
+            );
+            info!(
+                "Number of UnlockedVestingOutputFilter outcome outputs: {}",
+                self.number_of_outcome_outputs
+            );
+        }
+    }
 }
 
-fn scale_output_amount_for_iota(output: &mut Output) -> Result<()> {
-    *output = match output {
-        Output::Basic(ref basic_output) => {
-            // Update amount
-            let mut builder = BasicOutputBuilder::from(basic_output)
-                .with_amount(scale_amount_for_iota(basic_output.amount())?);
+/// Filter that looks for basic outputs having a tag being the Participation Tag
+/// and removes all features from the basic output. Operates in place during the
+/// iteration.
+#[derive(Default)]
+struct VotingOutputFilter {
```

Comment:
As mentioned before, it is not only "votes" but also "staking" transactions.
```suggestion
struct ParticipationOutputFilter {
```